### PR TITLE
manage nodeca sync daemon from the operator

### PR DESCRIPTION
**Pull request by bparees**
https://jira.coreos.com/browse/DEVEXP-237

Note that i am reusing the configmap that the registry is mounting (and that the registry operator is already maintaining, but i've added the service-ca to it).  So the set of additional CAs the registry trusts will be the same as the set of additional CAs the node trusts.  You can't set one without setting the other.  That was a more expedient approach than adding yet another config option for "what's the name of your node CA configmap", and it seemed reasonable to me that the registry and the node should be trusting the same set of CAs.

(if you want the node to pull from it, you probably want the registry to be able to pull-through from it)

### CODE REVIEW

File: pkg/resource/nodecadaemon.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	return ds.lister.Get(ds.GetName())
+}
+
+func (ds *generatorNodeCADaemonSet) Create() error {
+	daemonSet := resourceread.ReadDaemonSetV1OrDie([]byte(nodeCADaemonSetDefinition))
+	env := corev1.EnvVar{
+		Name:  "internalRegistryHostname",
+		Value: ds.hostname,
+	}
+	daemonSet.Spec.Template.Spec.Containers[0].Image = os.Getenv("IMAGE")
```
[the remark this answers is not in the document]
Comment (bparees) [reply pkg/resource/nodecadaemon.go]: thanks

**Comment by dmage** [comment]
We have some [problems with buckets finalization](https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_cluster-image-registry-operator/135/pull-ci-openshift-cluster-image-registry-operator-master-e2e-aws-operator/486):
```
	logs.go:77: I0106 07:28:33.048855       1 finalizer.go:51] finalizing *v1.ImageRegistry, Name=image-registry
	logs.go:77: E0106 07:28:33.241405       1 controller.go:215] unable to sync: unable to finalize resource: NoSuchBucket: The specified bucket does not exist
	logs.go:77: 	status code: 404, request id: 3BBD9ED11D03E7EA, host id: cqqVeafFCZ1LJjBm4VxrY97Cz5YeEmiCEkqQHrRp1GZzierA+Olv/ju5mNuc18nDJyQ9aA8MKNQ=, requeuing
```
/cc @coreydaley

**Comment by coreydaley** [comment]
@dmage I think that https://github.com/openshift/cluster-image-registry-operator/pull/137 will fix that.

**Comment by bparees** [comment]
> A little bit disturbing that Update cares only about internalRegistryHostname

@dmage not sure what you mean by that?

The changes to the CA values themselves are handled by k8s (we mount the configmap, if the configmap changes the pod will see the changes).  And the configmap itself is managed by the code you already put in place.

**Pull request merged**